fix: set correct permision UI during joining the community

mprakhov opened the pull request:
### What does the PR do

Fixes
- crash fix
- set correct permission required UI during joining the closed community
- if community permission is closed and the user owns TM or admin token - show this token in the join community permission view
- if all channels are hidden, show the "You don't hold the necessary tokens" UI only if the community is not closed and we fit the permissions to join the community
- do not show join community permission view if it is open community

Closes: 
- https://github.com/status-im/status-desktop/issues/14091
- https://github.com/status-im/status-desktop/issues/14085
- https://github.com/status-im/status-desktop/issues/14104

IMPACT:
- join community permission view for a member, who wants to join
- showing community channels permission

mprakhov committed:
fix: set correct permision UI

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 80f4634a | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-14156/1/) | 2024-03-25 23:01:29 | ~6 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-14156/1//consoleText) |

anastasiyaig commented:
@mprakhov what is correct UI here? Any figma link?

anastasiyaig commented:
Meanwhile i launched our community autotests and they are all good https://ci.status.im/job/status-desktop/job/e2e/job/manual/1705/allure/#suites/a9b9ce4ef70730fb5814a5009e98b445

@glitchminer maybe it could reduce some work for you

noeliaSD commented on ui/app/AppLayouts/Chat/views/ChatView.qml:
```diff
@@ -13,6 +13,7 @@ import shared.stores.send 1.0
 import SortFilterProxyModel 0.2
 
 import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
```
Is it needed?

noeliaSD commented on ui/imports/shared/stores/PermissionsStore.qml:
```diff
@@ -97,7 +97,9 @@ QtObject {
         function filterPredicate(modelData) {
-            return (modelData.permissionType == Constants.permissionType.member) &&
+            return (modelData.permissionType === Constants.permissionType.member ||
+                    modelData.permissionType === Constants.permissionType.admin ||
```
Why the `admin` role is added here if the model is about becoming member / joining community??

jrainville reviewed (approved):
Looks good

jrainville commented:
This will be needed in 2.28 after being tested and merged

noeliaSD commented:
> @anastasiyaig , @glitchminer waiting for other UI fixes from Noelia (she will merge them to this PR) and I think tomorrow this PR can be tested

I'll ping you when ready!

noeliaSD committed:
fix(JoinCommunity): Add missing state ONLY private permissions and NOT MET (#14164)

* fix(JoinCommunity): Add missing state ONLY private permissions and NOT MET

- Modified `becomeMemberModel` in store to provide all member permissions.
- Modified permissions model filter to only discard  permissions that are private and NOT met.
- Updated `storybook/PermissionsModel` with new only private permissions and added new model option in corresponding pages.

Closes #14104

* fix(JoinCommunity): Text position when all channels hidden

Updated text position when `allChannelsAreHiddenBecauseNotPermitted` in community join process

kounkou commented on storybook/pages/JoinCommunityPermissionsEditor.qml:
```diff
@@ -125,6 +125,16 @@ ColumnLayout {
         }
+
+        RadioButton {
+            text: "Private met model"
```
I am not sure what is a `Private met model` ?

noeliaSD replied:
It means a model where all the permission rows are private and at least one is met

glitchminer commented:
Hi @mprakhov , noted a couple of possible issues:

1. When the permissions of a channel are changed to hidden then the chat goes from the blurred permissions screen to being visible

https://github.com/status-im/status-desktop/assets/50769329/096737ae-a73f-4a0f-a5fa-18fdba1ae943

---
2. In the case where both "view and post" and "view" permissions are configured on a channel. If a member meets the requirements for "view and post" permissions but not "view" permissions then the user shown the chat and is told they must hold tokens that they already have if they want to post. Likely to be an edge case? 
![image](https://github.com/status-im/status-desktop/assets/50769329/5893c61f-0c22-4815-be46-78210449a9cb)
![image](https://github.com/status-im/status-desktop/assets/50769329/6d4760de-92ac-4f99-aa6a-dfb1eddad5a7)

---

Other stuff:
All community membership permissions hidden worked OK
Permissions screen not shown if no membership permissions set OK
Meeting requirements to be admin or TM while not meeting permissions to be a member worked OK

kounkou reviewed (approved):
Changes look good to me. 🚢  it!

anastasiyaig commented:
thanks @mprakhov and @glitchminer !
